numba/tests/test_fancy_indexing.py

```diff
@@ -301,5 +301,186 @@ def np_new_axis_setitem(a, idx, item):
             np.testing.assert_equal(expected, got)
 
 
+class TestFancyIndexingMultiDim(MemoryLeakMixin, TestCase):
+    # Every case has exactly one, one-dimensional array,
+    # otherwise it's not fancy indexing.
+    shape = (5, 6, 7, 8, 9, 10)
+    indexing_cases = [
+        # Slices + Integers
+        (slice(4, 5), 3, np.array([0, 1, 3, 4, 2]), 1),
+        (3, np.array([0,1,3,4,2]), slice(None), slice(4)),
+
+        # Ellipsis + Integers
+        (Ellipsis, 1, np.array([0,1,3,4,2])),
+        (np.array([0,1,3,4,2]), 3, Ellipsis),
+
+        # Ellipsis + Slices + Integers
+        (Ellipsis, 1, np.array([0,1,3,4,2]), 3, slice(1,5)),
+        (np.array([0,1,3,4,2]), 3, Ellipsis, slice(1,5)),
+
+        # Boolean Arrays + Integers
+        (slice(4, 5), 3,
+         np.array([True, False, True, False, True, False, False]),
+         1),
+        (3, np.array([True, False, True, False, True, False]),
+         slice(None), slice(4)),
+    ]
+
+    rng = np.random.default_rng(1)
```

Comment:
This needs to be in the function that uses it else the sequence generated depends on the order in which the tests are run? i.e. the test class itself is carrying state.